### PR TITLE
Add system.linux.memory.available metric

### LINKED ISSUE #257: Add system.linux.memory.available metric
**What are you trying to achieve?**

Linux kernel exposes a memory metric called `available`, which helps to understand how much memory is really available for the userspace apps. See https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7417.

This value is unique to Linux, hence the `system.linux` prefix: see https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/19149#issuecomment-1467634451.

**What did you expect to see?**

`system.linux.memory.available` - Gauge metric of integer type, meaning number of bytes available. No attributes.

**Additional context.**

https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7417#issuecomment-1668082340

cc @dmitryax @braydonk

### PR DESCRIPTION
**Pull request by ItsLastDay**
Fixes #257

## Changes

Add definition for a single metric `system.linux.memory.available`.

## Merge requirement checklist

* [yes ] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [yes] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* [not needed] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -21,18 +21,18 @@ key, but non-obvious, aspects:
 - All descriptions, normative language are defined in the `docs/`
   directory.
   - We provide tooling to generate Markdown documentation from the formal
-    YAML definitons.  See [Yaml to Markdown](#yaml-to-markdown).
+    YAML definitons. See [Yaml to Markdown](#yaml-to-markdown).
```
Comment (ChrsMark) [inline CONTRIBUTING.md:24]: Changes in this file seem to be unrelated?

File: model/metrics/system-metrics.yaml [hunk 2]
```diff
@@ -490,3 +490,13 @@ groups:
...
     instrument: counter
     unit: "{process}"
+
+  # system.linux.* metrics
+  - id: metric.system.linux.memory.available
+    type: metric
+    metric_name: system.linux.memory.available
+    brief: "An estimate of how much memory is available for starting new applications, without swapping"
+    note: |
+      Available since Linux 3.14: see `MemAvailable` in [here](https://man7.org/linux/man-pages/man5/proc.5.html).
```
Comment (ChrsMark) [inline model/metrics/system-metrics.yaml:500]: I see at https://github.com/open-telemetry/semantic-conventions/issues/257#issuecomment-1678926537 the background dicsussions on why to make this Linux specific.

However:
1) Do we have any plan/suggestion of what would be the equivalent for other operating systems? What we should suggest to our users to use instead of this?
2) If we finally keep the linux specific one we need to clearly state here in the [note the purpose of this metric and why it exists solely for Linux](https://github.com/open-telemetry/semantic-conventions/issues/257#issuecomment-1678926537).

File: CONTRIBUTING.md (hunk 1 above)
Comment (ItsLastDay) [reply CONTRIBUTING.md:24]: That's true. These changes are caused by `make fix` suggested [in contributing.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#autoformatting). 

Should I create a separate PR for that?

File: model/metrics/system-metrics.yaml (hunk 2 above)
Comment (ItsLastDay) [reply model/metrics/system-metrics.yaml:500]: 1. Just `state=free` value from https://github.com/open-telemetry/semantic-conventions/blob/47d7a1ed8080117b30020c949dddc4cff118298d/docs/system/system-metrics.md#metric-systemmemoryusage, nothing fancy
2. Ah, indeed, I did not do that. Can I just link https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/19149#issuecomment-1467634451 in the note? Or do I need to paste/rephrase it? (I guess the latter is better)

Comment (ChrsMark) [reply model/metrics/system-metrics.yaml:500]: I would mention the `state=free` alternative in the note along with a summary from https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/19149#issuecomment-1467634451. If we want to additionally add the link from the issue it would be fine but I think that the note should be enough on it's own (with or without the link).

File: CONTRIBUTING.md (hunk 1 above)
Comment (ChrsMark) [reply CONTRIBUTING.md:24]: Ah ok! I guess we are good to add those here since they are minor unless we have any objection from someone else.

Comment (ItsLastDay) [reply CONTRIBUTING.md:24]: > unless we have any objection from someone else.

I hope not :) because anyone who does `make fix` in any upcoming PRs would make the same changes.

File: model/metrics/system-metrics.yaml (hunk 2 above)
Comment (ItsLastDay) [reply model/metrics/system-metrics.yaml]: Got it. I updated the note, please take a look.

**Review by ChrsMark — APPROVED** [review]
LGTM.

File: docs/system/system-metrics.md [hunk 3]
```diff
@@ -739,3 +740,17 @@ an `{os}` prefix to split this metric across OSes.
...
+
+<!-- semconv metric.system.linux.memory.available(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `system.linux.memory.available` | UpDownCounter | `By` | An estimate of how much memory is available for starting new applications, without causing swapping [1] |
+
+**[1]:** This is an alternative to `system.memory.usage` metric with `state=free`.
+Linux starting from 3.14 exports "available" memory. It takes "free" memory as a baseline, and then factors in kernel-specific values.
+This is supposed to be more accurate than just "free" memory.
+People who are not experts in Linux Kernel are not expected to know the precise formula. For reference, see the calculations [here](https://superuser.com/a/980821).
```
Comment (joaopgrassi) [inline docs/system/system-metrics.md:754]: I'm not sure we need the "People who are not experts in Linux Kernel are not expected to know the precise formula" here. Maybe remove it and only leave the links for reference?

File: docs/system/system-metrics.md [hunk 4]
```diff
@@ -739,3 +740,17 @@ an `{os}` prefix to split this metric across OSes.
 
 [DocumentStatus]: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.22.0/specification/document-status.md
 [MetricRecommended]: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.22.0/specification/metrics/metric-requirement-level.md#recommended
+
+### Metric: `system.linux.memory.available`
```
Comment (joaopgrassi) [inline docs/system/system-metrics.md:744]: I think maybe for now is fine to leave this here, but if more OS's specific things pop up, we probably should create a new markdown file for it.

File: docs/system/system-metrics.md (hunk 3 above)
Comment (ItsLastDay) [reply docs/system/system-metrics.md]: Done, deleted

File: docs/system/system-metrics.md (hunk 4 above)
Comment (ItsLastDay) [reply docs/system/system-metrics.md:744]: acknowledged

**Review by joaopgrassi — APPROVED** [review]

**Comment by ChrsMark** [comment]
That will need a rebase to resolve the conflict in the changelog. @ItsLastDay could you please rebase?

**Comment by ItsLastDay** [comment]
Sure, done!

**Review by AlexanderWert — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Comment by ItsLastDay** [comment]
How do I merge this PR? Github says "The base branch restricts merging to authorized users. [Learn more about protected branches.](https://docs.github.com/articles/about-protected-branches/)". I don't understand what I am missing :\

**Comment by AlexanderWert** [comment]
> How do I merge this PR? Github says "The base branch restricts merging to authorized users. [Learn more about protected branches.](https://docs.github.com/articles/about-protected-branches/)". I don't understand what I am missing :\

Only @open-telemetry/specs-semconv-maintainers can merge
